filebeat/input/filestream/input.go

```diff
@@ -177,12 +190,14 @@ func initState(log *logp.Logger, c loginp.Cursor, s fileSource) state {
 	return state
 }
 
-func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, path string, offset int64) (reader.Reader, error) {
+func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, srcID, path string, offset int64) (reader.Reader, error) {
 	f, err := inp.openFile(log, path, offset)
 	if err != nil {
 		return nil, err
 	}
 
+	inp.startMonitoring(log, canceler, srcID, path, f)
```

Comment:
maybe we want to move `startMonitoring` to `Run`, in order to make it more obvious which tasks an input is running. This would require us to split `open` into `openFile` and `newReader`.